x-pack/legacy/plugins/apm/public/components/app/TraceLink/index.tsx

```diff
@@ -13,35 +13,44 @@ import { useFetcher, FETCH_STATUS } from '../../../hooks/useFetcher';
 import { useUrlParams } from '../../../hooks/useUrlParams';
 import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
 import { QueryParams } from '../../shared/Links/apm/ExternalLinks';
+import { TRACE_ID } from '../../../../common/elasticsearch_fieldnames';
 
 const CentralizedContainer = styled.div`
   height: 100%;
   display: flex;
 `;
 
+type DateRange = Partial<QueryParams>;
+
+const getDateRange = ({ rangeFrom, rangeTo }: DateRange) =>
+  rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {};
+
 const redirectToTransactionDetailPage = ({
   transaction,
   rangeFrom,
   rangeTo
-}: { transaction: Transaction } & Partial<QueryParams>) => {
-  const dateRange = rangeFrom && rangeTo ? { rangeFrom, rangeTo } : {};
-  return url.format({
+}: { transaction: Transaction } & DateRange) =>
+  url.format({
     pathname: `/services/${transaction.service.name}/transactions/view`,
     query: {
       traceId: transaction.trace.id,
       transactionId: transaction.transaction.id,
       transactionName: transaction.transaction.name,
       transactionType: transaction.transaction.type,
-      ...dateRange
+      ...getDateRange({ rangeFrom, rangeTo })
```

Comment:
What about simply doing this:

```js
query: {
      traceId: transaction.trace.id,
      transactionId: transaction.transaction.id,
      transactionName: transaction.transaction.name,
      transactionType: transaction.transaction.type,
      rangeFrom,
      rangeTo
}
```

Reply:
if the ranges are undefined the URL will be generated like `services/opbeans-ruby/transactions/view?rangeFrom=&rangeTo=&traceId=c130f46cb7e91675cf0be1a8fdd98dcc&transactionId=15e9fcc39fb5b4ce&transactionName=Api::OrdersController%23index&transactionType=request`

Reply:
When will this happen? The range is required in `getTraceUrl` afaict.
Even if it does happen - what's the consequence?

Reply:
no serious consequences